prov/gni/include/gnix_mr_cache.h

```diff
@@ -220,10 +221,18 @@ int _gnix_mr_cache_destroy(gnix_mr_cache_t *cache);
 int _gnix_mr_cache_flush(gnix_mr_cache_t *cache);
 
 /**
- * TODO
+ * @brief Initializes the MR cache state
+ *
+ * @param[in,out] cache   a gnix memory registration cache
+ * @param[in] mr_notifier notifier, @see struct gnix_mr_notifier
+ * @param[in] attr        cache attributes, @see gnix_mr_cache_attr_t
+ *
+ * @return           FI_SUCCESS on success
+ *                   -FI_ENOMEM otherwise
  */
 int _gnix_mr_cache_init(
 		gnix_mr_cache_t      **cache,
+		struct gnix_mr_notifier *mr_notifier,
```

Comment:
I'd like this to be a memory of the attributes that way if a notifier isn't specified, we can handle it in the init, rather than changing the API.